Better treatment of disputed lands

kent1D opened the pull request:
Borders should be dashed

A bit of OSX compatibility

Optimized svg icons

kent1D committed:
Verify we are using Python 3

For ESH, don't remove but add Area

kent1D committed:
Better for western Sahara

kent1D committed:
Optimized icons

kent1D committed:
Get the ISO code in the fields

kent1D committed:
sed --in-place doesn't work on mac, use sed -i.back and then remove the
temporary file

Keep the ISO code in the database

kent1D committed:
Better geojson

kent1D committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	admin.mss

kent1D committed:
Dashed border on disputed

yohanboniface commented on Makefile:
```diff
@@ -7,21 +7,23 @@ import:
 boundary:
 	python scripts/make_boundaries.py process
-	ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -select name,'name:en','name:fr','name:ar','name:es','name:de','name:ru' -nln itl_boundary
+	ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -select name,'name:en','name:fr','name:ar','name:es','name:de','name:ru','ISO3166-1:alpha2' -nln itl_boundary
```
I suggest we rename the column is "iso" here: `'ISO3166-1:alpha2' AS iso" should work.

kent1D replied:
100% for, I didn't know how to face it and didn't have the time to (or didn't want to, I don't know) read the doc :)

👍

kent1D replied:
Unfortunately it doesn't seem to work 

And this with different ways to quote.

I obtain : `ERROR 1: Couldn't fetch requested layer 'AS'!`

yohanboniface replied:
humm…
Have you tried `as`?

yohanboniface replied:
http://www.gdal.org/ogr_sql.html#ogr_sql_fname_alias

kent1D replied:
Seems to work as expected with :
```ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -sql 'SELECT name,"name:en","name:fr","name:ar","name:es","name:de","name:ru","ISO3166-1:alpha2" AS iso FROM OGRGeoJSON' -nln itl_boundary```

yohanboniface commented on admin.mss:
```diff
@@ -15,17 +16,19 @@
-  }
-  [zoom>=10] {
-    line-width: 2;
+  #itl_boundary[iso!='EH'], #itl_boundary_low[ISO3166-1!='EH']{
```
You don't need a subrule here, afaik.

yohanboniface reviewed (approved):
Sounds good, made quick comments about 'iso', but I can deal with them later also.

kent1D committed:
No need of a sub-rule here cc @yohanboniface

Also :
- disputed borders should be visible from 0
- disputed borders should be visible for every zoom level
- disputed pattern until zoom 12

yohanboniface commented on admin.mss:
```diff
@@ -1,34 +1,34 @@
-    [zoom>=10] {
-      line-width: 2;
-    }
+#itl_boundary_low[ISO3166-1!='EH'][zoom>=1][zoom<6][maritime!='yes'],
```
Thinking about it a second time, why not filtering this out during the make_boundary step?

yohanboniface replied:
Third though, this should not even be needed, given we don't have the Western Sahara in our CSV. Which at the end is all the list of country we process.

(And BTW, it's not a `boundary=administrative` but `boundary=disputed`: https://www.openstreetmap.org/relation/2559126)

kent1D replied:
Of course you are right, this was the problem

https://github.com/tilery/pianoforte/pull/20/commits/4f2dcde635ce882c8dd7fc38e49d96a4b17154b8#diff-1b8328945b14d8c22b127c55a04b3091L171

kent1D committed:
Import directly the iso code as iso

renaming .back => .bak for consistency

kent1D commented:
```
ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -sql 'SELECT name,"name:en","name:fr","name:ar","name:es","name:de","name:ru","ISO3166-1:alpha2" AS iso FROM OGRGeoJSON' -nln itl_boundary
```

Seems to work well

kent1D committed:
For small zoom levels, divide by 2 the pattern

kent1D committed:
Do not append western sahara here

kent1D committed:
Of course we don't need this negative selector, since we shouldn't have
it in datas

kent1D committed:
And the simplified version

yohanboniface commented on Makefile:
```diff
@@ -7,21 +7,23 @@ import:
 boundary:
 	python scripts/make_boundaries.py process
-	ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -select name,'name:en','name:fr','name:ar','name:es','name:de','name:ru' -nln itl_boundary
+	ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -sql 'SELECT name,"name:en","name:fr","name:ar","name:es","name:de","name:ru","ISO3166-1:alpha2" AS iso FROM OGRGeoJSON' -nln itl_boundary
```
Humm, which ogr2ogr version are you running?
Table is name "boundary" for me, not "OGRGeoJSON".

```
ogr2ogr --config PG_USE_COPY YES -lco GEOMETRY_NAME=geometry -lco DROP_TABLE=IF_EXISTS -f PGDump tmp/boundary.sql data/boundary.json -sql 'SELECT name,"name:en","name:fr","name:ar","name:es","name:de","name:ru","ISO3166-1:alpha2" AS iso FROM OGRGeoJSON' -nln itl_boundary
ERROR 1: SELECT from table OGRGeoJSON failed, no such table/featureclass.
```

kent1D replied:
```
$ ogr2ogr --version
GDAL 2.1.3, released 2017/20/01
```

I tried with that, but didn't work and looked at tutorials.

```
$ ogrinfo data/boundary.json
INFO: Open of `data/boundary.json'
      using driver `GeoJSON' successful.
1: OGRGeoJSON
```

yohanboniface replied:
Ah, from http://www.gdal.org/drv_geojson.html

> Starting with GDAL 2.2, the layer name is built with the following logic:

>    If a "name" member is found at the FeatureCollection level, it is used.
>    Otherwise if the filename is regular (ie not a URL with query parameters), then the filename without extension and path is used as the layer name.
>    Otherwise OGRGeoJSON is used.

Can you upgrade so we align on latest? Or should I downgrade? :)

kent1D replied:
To avoid this, we may add a name attribute to the FeatureCollection ?

https://github.com/tilery/pianoforte/blob/master/scripts/make_boundaries.py#L185

yohanboniface replied:
For the record:

```
$ ogrinfo data/boundary.json
INFO: Open of `data/boundary.json'
      using driver `GeoJSON' successful.
1: boundary
```

yohanboniface replied:
> To avoid this, we may add a name attribute to the FeatureCollection ?

Will this work with GDAL < 2.2?

yohanboniface replied:
(Can you try? :) )

kent1D replied:
here : http://www.gdal.org/drv_geojson.html

Starting with GDAL 2.2, the layer name is built with the following logic:

* If a "name" member is found at the FeatureCollection level, it is used.
* Otherwise if the filename is regular (ie not a URL with query parameters), then the filename without extension and path is used as the layer name.
* Otherwise OGRGeoJSON is used.

kent1D replied:
No, unforunately doesn't work.

I will try to upgrade

kent1D replied:
Ok, Gdal 2.2.x needed cf : #21

kent1D commented:
```
$ ogr2ogr --version
GDAL 2.1.3, released 2017/20/01
```

I tried with that, but didn't work and looked at tutorials.

```
$ ogrinfo data/boundary.json
INFO: Open of `data/boundary.json'
      using driver `GeoJSON' successful.
1: OGRGeoJSON
```
